Tests/ML/models/test_scalar_model.py

```diff
@@ -337,7 +346,36 @@ def test_runner1(test_output_dirs: OutputFolderForTests) -> None:
     assert config.non_image_feature_channels == ["label"]
     assert str(config.outputs_folder).startswith(output_root)
     assert (config.logs_folder / LOG_FILE_NAME).exists()
+    # Check that we saved one checkpoint every second epoch and that we kept only that last 2 and that last.ckpt has
+    # been renamed to best.ckpt
+    assert len(os.listdir(config.checkpoint_folder)) == 3
+    assert (config.checkpoint_folder / str(RECOVERY_CHECKPOINT_FILE_NAME + "_epoch=3" + CHECKPOINT_SUFFIX)).exists()
+    assert (config.checkpoint_folder / str(RECOVERY_CHECKPOINT_FILE_NAME + "_epoch=5" + CHECKPOINT_SUFFIX)).exists()
+    assert (config.checkpoint_folder / BEST_CHECKPOINT_FILE_NAME_WITH_SUFFIX).exists()
+
 
+@pytest.mark.skipif(common_util.is_windows(), reason="Has OOM issues on windows build")
+def test_runner_restart(test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test if starting training from a folder where the checkpoints folder already has recovery checkpoints picks up
+    that it is a recovery run. Also checks that we update the start epoch in the config at loading time.
+    """
+    model_config = DummyClassification()
+    model_config.set_output_to(test_output_dirs.root_dir)
+    model_config.num_epochs = 6
+    model_config.recovery_checkpoint_save_interval = 2
+    model_config.save_last_k_recovery_checkpoints = -1
+    runner = MLRunner(model_config=model_config)
+    runner.setup(use_mount_or_download_dataset=False)
+    shutil.copytree(full_ml_test_data_path("recovery_dummy_checkpoints"), runner.container.checkpoint_folder)
```

Comment:
We completely switched away from having stored checkpoints, to simplify long-term maintainability. Better to use `create_model_and_store_checkpoint` and create the checkpoints on-the-fly.